doc/nrf/ug_bootloader_external_flash.rst

```diff
@@ -13,57 +12,14 @@ See :ref:`pm_external_flash` for general information about how to set up partiti
-For MCUboot, this configuration is :file:`ncs/bootloader/mcuboot/boot/zephyr/pm.yml`.
+To use the external flash memory as storage for the secondary partition, you must manually specify the size of the secondary partition by setting the value of the :kconfig:`CONFIG_PM_PARTITION_SIZE_MCUBOOT_SECONDARY` configuration option to match the value of the automatically deduced size of the primary partition.
+CMake gives a warning when the values do not match, and it also provides the correct value.
 
-In this file, you must define the region under the section ``mcuboot_secondary``:
-
-.. code-block:: yaml
-
-   mcuboot_secondary:
-       region: external_flash
-       size: CONFIG_PM_EXTERNAL_FLASH_SIZE
-
-The nRF52840 DK comes with an external flash memory that can be used for the secondary slot and be accessed using the QSPI NOR flash memory driver.
-The following steps show how to configure an application for the nRF52840 DK:
-
-1. Add the following configuration options to the ``prj.conf`` file associated with the MCUboot child image:
-
-   .. code-block:: console
-
-      CONFIG_NORDIC_QSPI_NOR=y
-      CONFIG_NORDIC_QSPI_NOR_FLASH_LAYOUT_PAGE_SIZE=4096
-      CONFIG_NORDIC_QSPI_NOR_STACK_WRITE_BUFFER_SIZE=4
-      CONFIG_MULTITHREADING=y
-      CONFIG_BOOT_MAX_IMG_SECTORS=256
-      CONFIG_PM_EXTERNAL_FLASH=y
-      CONFIG_PM_EXTERNAL_FLASH_DEV_NAME="MX25R64"
-      CONFIG_PM_EXTERNAL_FLASH_SIZE=0xf4000
-      CONFIG_PM_EXTERNAL_FLASH_BASE=0
-
-   These options enable the QSPI NOR flash memory driver, multi-threading (required by the flash memory driver), and the external flash memory of the nRF52840 DK.
-
-#. Update the :file:`ncs/bootloader/mcuboot/boot/zephyr/pm.yml` file as mentioned above:
-
-   .. code-block:: yaml
-
-      mcuboot_secondary:
-          region: external_flash
-          size: CONFIG_PM_EXTERNAL_FLASH_SIZE
-
-#. Add the following configuration options to the :file:`prj.conf` file in your application directory:
-
-   .. code-block:: console
+.. note::
 
-      CONFIG_NORDIC_QSPI_NOR=y
-      CONFIG_NORDIC_QSPI_NOR_FLASH_LAYOUT_PAGE_SIZE=4096
-      CONFIG_NORDIC_QSPI_NOR_STACK_WRITE_BUFFER_SIZE=4
-      CONFIG_PM_EXTERNAL_FLASH=y
-      CONFIG_PM_EXTERNAL_FLASH_DEV_NAME="MX25R64"
-      CONFIG_PM_EXTERNAL_FLASH_SIZE=0xf4000
-      CONFIG_PM_EXTERNAL_FLASH_BASE=0
+    Currently, you must manually synchronize the value of :kconfig:`CONFIG_PM_PARTITION_SIZE_MCUBOOT_SECONDARY` by setting it for both the parent image and the MCUboot child image.
 
-   These options enable the QSPI NOR flash memory driver and the external flash memory of the nRF52840 DK.
-   Multi-threading is enabled by default, so you do not need to enable it again.
+Both the nRF52840 DK and nRF5340 DK come with an external flash memory that can be used for the secondary slot and can be accessed using the QSPI NOR flash memory driver.
```

Comment:
Can the external flash be used for the secondary slot, and also used for a full modem update?

Reply:
Currently we only support QSPI - connected external flash for use as mcuboot secondary slot out of the box. The nRF91 only has SPI, so there we don't provide any ready-to-go solution for storing the secondary slot in external flash.

That being said: With some tweaking there is no problem storing both mcuboot secondary and the full modem update in an external flash on the nRF91.